### DISCUSSION TITLE
Follow all redirects, including HTML meta and Javascript

### DISCUSSION THREAD

**Discussion by rzb** [spatie/crawler#424] [Q&A]
I need to report the whole redirect chain (HTTP, meta refresh and javascript) in the same order that a human user would experience. I see that you use a headless browser to execute js, so is it something this package can do?

**Comment by bessone** [comment]
I have the same question, now I use Guzzle to read redirects, but obviously meta and javascript are not considered.

I found a discussion to do this correctly with Puppeteer (https://github.com/puppeteer/puppeteer/issues/2163), but in my project I don't use Node.js directly and doing it directly in PHP would be perfect!
